CONTRIBUTING.md

```diff
@@ -0,0 +1,88 @@
+# Contributing Pulumi Documentation
+
+## Pulumi terminology
+
+- Use `pulumi` or "the Pulumi CLI" to refer to the CLI.
+- Use `pulumi.com` to refer to the service.
+
+## Documentation structure
+
+- There is a folder for each heading in the top navigation, such as `Install`, `getting-started`, etc.
+
+- The mapping from documentation page to section and table-of-contents (TOC) is stored largely in each page's front matter, leveraging [Hugo Menus](https://gohugo.io/content-management/menus/). Menus for the CLI commands and API reference are specified in `./config.toml`.
+
+### Links to other files
+
+We generally use Hugo's [`relref` shortcode](https://gohugo.io/content-management/shortcodes/#ref-and-relref) when linking to other pages. Examples:
+
+```markdown
+[Install]({{< relref "/docs/get-started/install" >}})
+[Outputs]({{< relref "/docs/intro/concepts/stack#outputs" >}})
+```
+
+Which, on a page inside the `./content/reference` directory, will generate:
+
+```html
+<a href="/docs/install/">Install</a>
+<a href="/docs/intro/concepts/stack/#stack-outputs">Outputs</a>
+```
+
+### Hugo tips
+
+- **Redirects.** If you rename a file or directory, add a 301 redirect in the front-matter via an [alias](https://gohugo.io/content-management/urls/#aliases) `aliases: [/previous-dir/previousfile.html]`.
+
+- **Includes.**
+
+  - **.md files.** To share common content across articles, use [Hugo Shortcodes](https://gohugo.io/content-management/shortcodes/). Place a .html file in the [layouts/shortcodes] folder. To include it in a page, use syntax `{{< my-shortcode >}}`
+
+    For example, our custom [`cleanup`](layouts/shortcodes/cleanup.html) shortcode can be included in .md files, to include common text about cleaning up stack resources:
+
+    ```md
+    {{< cleanup >}}
+    ```
+
+  - **.html layout files.** HTML layouts can include other layouts inside the [layouts/partials](layouts/partials) directory, e.g.:
+
+    ```html
+    {{ partial "head.html" . }}
+    ```
+
+- **Front-matter variables.** You can define a front-matter variable in the YAML section at the top of a file. For instance, the you could add the following front matter `foo: "bar"`, and then reference the variable in markdown with the syntax `{{< param foo >}}`.
+
+  - **no_on_this_page** Specify this variable to prevent displaying an "On This Page" TOC on the right nav for the page.
+  - **block_external_search_index** Specify this variable to prevent crawlers from indexing the page.
+## Style guide
```

Comment:
for this section might be cool to get anything not in the hugo style guide there, and then link to it?

Reply:
For sure, yeah. We can treat that new doc as the canonical one now, and then link all others to it. (This WIP predates that doc, and this file (CONTRIBUTING.md) is just a straight-up copy from [pulumi/docs/CONTRIBUTING.md](https://github.com/pulumi/docs/blob/master/CONTRIBUTING.md) that I was planning on trimming down to just what makes sense to have in this repo.)